### ISSUE TITLE
Carts being merged with carts that have orders

### ISSUE THREAD

**Issue by ryanmitchell** [lunarphp/lunar#752]
We are seeing a strange issue when allowing carts to be merged on login... users are then placing payments a few minutes later with the same cart items (and others added), but the order doesnt come through as the cart is already associated with an order.

We are using CartSession::forget() so the only thing I can put it down to is different devices and carts being merged on login.

I think the issue is that the queries in:
https://github.com/lunarphp/lunar/blob/main/packages/core/src/Actions/Carts/AssociateUser.php
don't check if the Cart is `active()`

Should they not be scoped that way?

**Comment by alecritson** [comment]
Thanks @ryanmitchell Yeah it does feel like we should certainly be checking that it's an active cart before it's merged.

**Comment by alecritson** [comment]
@ryanmitchell Just to clarify, are you running your site on the `main` branch? i.e. `0.2` ?

**Comment by ryanmitchell** [comment]
No, running it on the latest release version. Why is there something in the main branch thats related?

**Comment by alecritson** [comment]
There's changes to the way carts work in `0.2` and the file you referenced is new to `0.2` so will need patching out on `0.1` also :)
